rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -149,9 +151,9 @@ where
         .collect::<Vec<(u32, T::Native)>>();
     let mut nulls = null_indices;
     if !options.descending {
-        valids.sort_by_key(|a| a.1);
+        valids.sort_by(|a, b| a.1.partial_cmp(&b.1).unwrap());
```

Comment:
Won't this panic if the vector contains any `NaN` values? partial_cmp would return `None` in that case.

Reply:
Hmm, do we ever store NaN, or do we represent it as a null? I've never tried creating an array with NaN